action.yml

```diff
@@ -20,33 +20,33 @@ inputs:
     default: ''
 
 runs:
-  using: composite
+  using: 'composite'
   steps:
-    - name: Add .ssh directory
+    - name: 'Add .ssh directory'
       run: |
         mkdir -p ~/.ssh
         chmod 700 ~/.ssh
       shell: bash
 
-    - name: Copy private key
+    - name: 'Copy private key'
+      shell: 'bash'
       run: |
         echo "${{ inputs.private-key }}" > ~/.ssh/${{ inputs.private-key-name }}
         chmod 400 ~/.ssh/${{ inputs.private-key-name }}
-      shell: bash
 
     - name: Copy config
-      if: inputs.ssh-config != ''
+      if: '${{ inputs.ssh-config != "" }}'
```

Comment:
I'm not sure!
I always write `if: "${{ inputs.ssh-config != '' }}"`

Reply:
> When you use expressions in an if conditional, you may omit the expression syntax (${{ }}) because GitHub automatically evaluates the if conditional as an expression.

Source : [GitHub Docs](https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#runsstepsif)

It's valid, but maybe more common to see the expression syntax `${{ }}`. I'm ok to change.
Quotes are probably not needed, I don't see any example from the docs using quotes to wrap the expression.

Reply:
> Quotes are probably not needed,

Quotes are for YAML strings.

I leave is 100% up to you.
My policy is to be always full explicit. No implicit hidden features.
https://github.com/szepeviktor/debian-server-tools/blob/master/webserver/Continuous-integration-Continuous-delivery.yml

Reply:
- expressions are necessary for the computer to run
- `${{ }}` are for the reader
- YAML quotes are for protecting my ass

Reply:
> To use a literal single quote, escape the literal single quote using an additional single quote (''). Wrapping with double quotes (") will throw an error.

😭